File: algorithm/Greedy/boj2217로프.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Collections;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		int n = Integer.parseInt(st.nextToken());
		int[] arr = new int[n];
		
		for (int i=0; i<n; i++) {
			st = new StringTokenizer(br.readLine());
			arr[i] = Integer.parseInt(st.nextToken());
		}
		
		Arrays.sort(arr, Collections.reverseOrder());
		
		int[] result = new int[n];
		for (int j=0; j<n; j++) {
			result[j] = arr[j] * (j+1);
		}
		
		int mx = 0;
		for (int k=0; k<result.length; k++) {
			mx = Math.max(mx, result[k]);
		}
		System.out.println(mx);
	}
}

File: algorithm/Greedy/boj 2217 로프.py
n = int(input())

rope = []
for _ in range(n):
    rope.append(int(input()))

rope.sort(reverse=True)
result = []
for i in range(n):
    result.append(rope[i]*(i+1))
print(max(result))
'''
30 27 25 15

1개 -> 30
2개 -> 27*2
3개 -> 25*3
4개 -> 15*4
'''
